### ISSUE TITLE
Ruff formatter adds spaces around colons in complex slice expressions, violating PEP 8

### ISSUE THREAD

**Issue by ten24bytes** [astral-sh/ruff#20530]
## Description

Ruff's formatter incorrectly adds spaces around colons in complex slice expressions (e.g., `i : i + batch_size`), which violates PEP 8 guidelines. According to PEP 8, slice notation should not have spaces around the colon.

---

## Expected Behavior

According to PEP 8:

> "Always surround these binary operators with a single space on either side... However, in a slice the colon acts like a binary operator, and should have equal amounts on either side (treating it as the operator with the lowest priority). In an extended slice, both colons must have the same amount of spacing applied."

For slice notation, this means **no spaces around colons**:

```python
list[start:end:step]
```

---

## Actual Behavior

Ruff formatter adds spaces in complex slice expressions but not in simple ones, creating inconsistent formatting:

* ✅ Simple slices formatted correctly:

  ```python
  data[0:5]
  ```
* ❌ Complex slices formatted incorrectly:

  ```python
  data[i : i + batch_size]  # should be data[i:i + batch_size]
  ```

---

## Reproduction Steps

Create a Python file with the following content:

```python
# test_slice_format.py
data = list(range(100))
batch_size = 10

# Simple slice - formats correctly
simple = data[0:5]

# Complex slice - formats incorrectly
for i in range(0, len(data), batch_size):
    batch = data[i:i + batch_size]

# More examples
result = data[start:end]
result2 = data[i:i + 1]
result3 = data[index:index + size]
```

Run:

```bash
ruff format test_slice_format.py
```

Observe that complex slices get formatted with spaces:

```python
data[i : i + batch_size]
```

---

## Environment

* **Ruff version:** >=0.12.8
* **Python version:** 3.13
* **OS:** Windows
* **Configuration:** Standard configuration with no special slice-related settings

```toml
[tool.ruff]
target-version = "py313"

[tool.ruff.format]
quote-style = "double"
indent-style = "space"
skip-magic-trailing-comma = false
line-ending = "auto"
docstring-code-format = false
docstring-code-line-length = "dynamic"
```

---

## Impact

This inconsistency:

* Violates PEP 8 compliance
* Creates inconsistent code formatting within the same file
* Requires manual workarounds using `# fmt: skip` comments
* Makes it difficult to maintain consistent code style in projects

---

## Workaround

Currently using `# fmt: skip` comments to prevent incorrect formatting:

```python
batch = data[i:i + batch_size]  # fmt: skip
```

---

## Additional Context

This issue appears to affect **complex slice expressions** where the slice indices involve arithmetic operations or variable concatenation.

* ✅ Simple literal slices (like `[0:5]`) are formatted correctly.
* ❌ Complex slices (like `[i:i + batch_size]`) are reformatted incorrectly.

The inconsistency suggests that Ruff's parser treats complex slice expressions differently from simple ones, possibly applying **general binary operator spacing rules** instead of **slice-specific formatting rules**.

### Version

ruff 0.13.1

**Comment by MichaReiser** [comment]
I don't think this violates PEP8. From PEP8

```py
# Correct:
ham[1:9], ham[1:9:3], ham[:9:3], ham[1::3], ham[1:9:]
ham[lower:upper], ham[lower:upper:], ham[lower::step]
ham[lower+offset : upper+offset]
ham[: upper_fn(x) : step_fn(x)], ham[:: step_fn(x)] # <-- This
ham[lower + offset : upper + offset] # <-- This
```

also note:

>  and should have equal amounts on either side (treating it as the operator with the lowest priority). In an extended slice, both colons must have the same amount of spacing applied.
